### PR TITLE
CMake changes for Ninja

### PR DESCRIPTION
**Pull request by umar456**

### CODE REVIEW

File: CMakeModules/build_cl2hpp.cmake [hunk 1]
```diff
@@ -9,6 +9,7 @@ ExternalProject_Add(
     PREFIX "${prefix}"
     INSTALL_DIR "${prefix}/package"
     UPDATE_COMMAND ""
+    BUILD_BYPRODUCTS "${prefix}/package/CL/cl2.hpp"
```
Comment (pavanky) [inline CMakeModules/build_cl2hpp.cmake]: This is breaking. Is this `BUILD_BYPRODUCTS` or `BYPRODUCTS` ?

Comment (umar456) [inline CMakeModules/build_cl2hpp.cmake]: Its BUILD_BYPRODUCTS for this command. I guess it is a newer CMAKE option. I'll take a look.

**Comment by 9prady9** [comment]
@umar456 
```diff
diff --git a/CMakeModules/build_cl2hpp.cmake b/CMakeModules/build_cl2hpp.cmake
index 9c950af..0cf2b49 100644
--- a/CMakeModules/build_cl2hpp.cmake
+++ b/CMakeModules/build_cl2hpp.cmake
@@ -2,6 +2,15 @@ INCLUDE(ExternalProject)
 
 SET(prefix ${PROJECT_BINARY_DIR}/third_party/cl2hpp)
 
+IF(CMAKE_VERSION VERSION_LESS 3.2)
+    IF(CMAKE_GENERATOR MATCHES "Ninja")
+        MESSAGE(WARNING "Building forge with Ninja has known issues with CMake older than 3.2")
+    endif()
+    SET(byproducts)
+ELSE()
+    SET(byproducts BYPRODUCTS "${prefix}/package/CL/cl2.hpp")
+ENDIF()
+
 ExternalProject_Add(
     cl2hpp-ext
     GIT_REPOSITORY https://github.com/KhronosGroup/OpenCL-CLHPP.git
@@ -17,6 +26,7 @@ ExternalProject_Add(
     -DBUILD_DOCS:BOOL=OFF
     -DBUILD_EXAMPLES:BOOL=OFF
     -DBUILD_TESTS:BOOL=OFF
+    ${byproducts}
     )
 
 ExternalProject_Get_Property(cl2hpp-ext install_dir)
```
This seems to work and it is similar to other `build_*` scripts we have under `CMakeModules` folder.

**Comment by umar456** [comment]
@9prady9 That worked. Much better now. Thanks!

**Comment by umar456** [comment]
build arrayfire windows ci

**Comment by umar456** [comment]
build arrayfire windows ci

**Pull request merged**